qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -851,23 +849,28 @@ def add_collection(self, data: Dict[str, Any], overwrite: bool = False):
         return ret
 
     def get_collections(self,
-                        collection: str = None,
-                        name: str = None,
-                        return_json: bool = True,
-                        with_ids: bool = True,
-                        limit: int = None,
-                        projection: Dict[str, Any] = None,
+                        collection: Optional[str] = None,
+                        name: Optional[str] = None,
+                        col_id: Optional[int] = None,
+                        limit: Optional[int] = None,
+                        projection: Optional[Dict[str, Any]] = None,
```

Comment:
This can cause a big problem if it's called with no parameters, `get_collection()`. The use of no filters at all should raise an exception. No `get_` method should allow getting all the DB unintentionally.
However, `get_collection(collection=collection_type)` is okay.

Reply:
We do need the ability to call `get_collections` without a query, but with a projection which is how the client side `list_collections` work. IIRC `get_molecules` and the like are uncapped, up for changing this where we add an explicit `list_collections` storage socket function as well.

Reply:
I'm going to make an issue and kick this down the road.